Remove the unindent for nested Markdown

justincampbell opened the pull request:
When nesting a Markdown code block, the leading whitespace was stripped regardless of the context.

Example of what is was doing before:

![](http://c.justincampbell.me/221t0N1K0Y3z/Screen%20Shot%202017-10-20%20at%201.37.25%20PM.png)
![](http://c.justincampbell.me/2A1R07372g2M/Screen%20Shot%202017-10-20%20at%201.36.42%20PM.png)

sethvargo commented:
Hey @justincampbell 

Can you just verify on a few other pages that things look okay? I _think_ this will be okay since the tests are happy, but a quick visual click-through would make me feel better.

justincampbell commented:
Sure! But just to clarify, I'd have to pick a sampling of other projects (Nomad/Consul/Terraform/etc), build them locally with this change, and see if anything broke?

sethvargo commented:
@justincampbell yea 😦 . It's sad. The process is:

1. Download project, `cd website` (don't use terraform, it's special)
1. Update gemfile to point to local path
1. Run `bundle` and `bundle exec middleman`

justincampbell commented:
Going to run `middleman build` and diff the changes.

justincampbell commented:
### Vagrant

```diff
diff --git a/website/build/docs/cli/machine-readable.html b/website/build/docs/cli/machine-readable.html
index 8fc10ba99..1ae876c65 100644
--- a/website/build/docs/cli/machine-readable.html
+++ b/website/build/docs/cli/machine-readable.html
@@ -664,7 +664,7 @@ The provider name of the target machine.
 <td>ssh-config</td>
 <td>
 The OpenSSH compatible SSH config for a machine. This is usually
-the result of the &quot;ssh-config&quot; command.
+  the result of the &quot;ssh-config&quot; command.
 <span class="label">targeted</span>
 </td>
 </tr></p>
diff --git a/website/build/docs/provisioning/cfengine.html b/website/build/docs/provisioning/cfengine.html
index 23e6d48fd..d85cb5c25 100644
--- a/website/build/docs/provisioning/cfengine.html
+++ b/website/build/docs/provisioning/cfengine.html
@@ -567,7 +567,7 @@ and the clients in a single
 <p><strong>Warning:</strong> If you are not familiar with CFEngine and Vagrant already,
 I recommend starting with the <a href="/docs/provisioning/shell.html">shell
 provisioner</a>. However, if you are comfortable with Vagrant
-already, Vagrant is the best way to learn CFEngine.</p>
+  already, Vagrant is the best way to learn CFEngine.</p>
 </div><p>Let us look at some common examples first. See the bottom of this
 document for a comprehensive list of options.</p>
 <h2 id="setting-up-a-cfengine-server-and-client">
diff --git a/website/build/docs/provisioning/docker.html b/website/build/docs/provisioning/docker.html
index 6f71de799..f17884ff4 100644
--- a/website/build/docs/provisioning/docker.html
+++ b/website/build/docs/provisioning/docker.html
@@ -574,7 +574,7 @@ your application runtime. You can use the Puppet provisioner along
 with the Docker provisioner.</p>
 <div class="alert alert-info">
 <p><strong>Note:</strong> This documentation is for the Docker
-<em>provisioner</em>. If you are looking for the Docker
+    <em>provisioner</em>. If you are looking for the Docker
 <em>provider</em>, visit the
 <a href="/docs/docker/">Docker provider documentation</a>.</p>
 </div><h2 id="options">
diff --git a/website/build/docs/synced-folders/nfs.html b/website/build/docs/synced-folders/nfs.html
index 26a034d54..d01507876 100644
--- a/website/build/docs/synced-folders/nfs.html
+++ b/website/build/docs/synced-folders/nfs.html
@@ -677,7 +677,7 @@ are located in non-standard paths, modify them as appropriate.</p>
 <p>All of the snippets below require Vagrant version 1.7.3 or higher.</p>
 <div class="alert alert-warning" role="alert">
 <p><strong>Use the appropiate group for your user</strong> Depending on how your machine is
-configured, you might need to use a different group than the ones listed in the examples below.</p>
+ configured, you might need to use a different group than the ones listed in the examples below.</p>
 </div><p>For OS X, sudoers should have this entry:</p>
 <div class="highlight"><pre class="highlight plaintext"><code>Cmnd_Alias VAGRANT_EXPORTS_ADD = /usr/bin/tee -a /etc/exports
 Cmnd_Alias VAGRANT_NFSD = /sbin/nfsd restart
```

justincampbell commented:
### Nomad

Has a number of changes to tables similar to this:

```diff
diff --git a/website/build/docs/agent/configuration/acl.html b/website/build/docs/agent/configuration/acl.html
index b825f3aeb..bd5405717 100644
--- a/website/build/docs/agent/configuration/acl.html
+++ b/website/build/docs/agent/configuration/acl.html
@@ -813,10 +813,10 @@
 </h1>
 <table class="table table-bordered table-striped">
 <p><tr>
-<th width="120">Placement</th>
-<td>
-  <code><strong>acl</strong></code>
-</td>
+  <th width="120">Placement</th>
+  <td>
+    <code><strong>acl</strong></code>
+  </td>
 </tr></p>
 </table><p>The <code>acl</code> stanza configures the Nomad agent to enable ACLs and tunes various ACL parameters.</p>
 <div class="highlight"><pre class="highlight hcl"><code><span class="n">acl</span> <span class="p">{</span>
diff --git a/website/build/docs/agent/configuration/client.html b/website/build/docs/agent/configuration/client.html
index 3a97894a3..80530fef7 100644
--- a/website/build/docs/agent/configuration/client.html
+++ b/website/build/docs/agent/configuration/client.html
@@ -813,10 +813,10 @@
 </h1>
 <table class="table table-bordered table-striped">
 <p><tr>
-<th width="120">Placement</th>
-<td>
-  <code><strong>client</strong></code>
-</td>
+  <th width="120">Placement</th>
+  <td>
+    <code><strong>client</strong></code>
+  </td>
 </tr></p>
 </table><p>The <code>client</code> stanza configures the Nomad agent to accept jobs as assigned by
 the Nomad server, join the cluster, and specify driver-specific configuration.</p>
```

Does not seem to affect the display:

![](http://c.justincampbell.me/1k3K342v0C2v/Screen%20Shot%202017-10-20%20at%202.35.51%20PM.png)

> (Difference in perceived font weight I believe is due to font loading local vs official)

justincampbell commented:
### Consul

```diff
diff --git a/website/build/api/libraries-and-sdks.html b/website/build/api/libraries-and-sdks.html
index e2bde730..a18239e8 100644
--- a/website/build/api/libraries-and-sdks.html
+++ b/website/build/api/libraries-and-sdks.html
@@ -391,68 +391,68 @@ more conveniently. Some are officially maintained while others are provided by
 the community.</p>
 <ul>
 <p><li>
- <a href="https://github.com/hashicorp/consul/tree/master/api">api</a> - Official Go client for the Consul HTTP API
+  <a href="https://github.com/hashicorp/consul/tree/master/api">api</a> - Official Go client for the Consul HTTP API
 </li>
 <li>
- <a href="https://github.com/gmr/consulate">consulate</a> - Python client for the Consul HTTP API
+  <a href="https://github.com/gmr/consulate">consulate</a> - Python client for the Consul HTTP API
 </li>
 <li>
- <a href="https://github.com/cablehead/python-consul">python-consul</a> - Python client for the Consul HTTP API
+  <a href="https://github.com/cablehead/python-consul">python-consul</a> - Python client for the Consul HTTP API
 </li>
```

![](http://c.justincampbell.me/2w2M0F2D0f0z/Screen%20Shot%202017-10-20%20at%202.46.20%20PM.png)

```diff
diff --git a/website/build/docs/agent/rpc.html b/website/build/docs/agent/rpc.html
index 52cfff14..1178d647 100644
--- a/website/build/docs/agent/rpc.html
+++ b/website/build/docs/agent/rpc.html
@@ -821,12 +821,12 @@ response body looks like:</p>
     <span class="p">...</span>
   <span class="p">},</span>
   <span class="s2">"consul: {
-    "</span><span class="nx">server</span><span class="s2">": "</span><span class="kc">true</span><span class="err">"</span><span class="p">,</span>
-    <span class="p">...</span>
-  <span class="p">},</span>
-  <span class="p">...</span>
-<span class="p">}</span>
-</code></pre></div><h3 id="reload">
+    "</span><span class="nx">server</span><span class="s2">": "</span><span class="kc">true</span><span class="s2">",
+    ...
+  },
+  ...
+}
+</span></code></pre></div><h3 id="reload">
   <a name="reload" class="anchor" href="#reload">&raquo;</a>
   reload
 </h3>
```

![](http://c.justincampbell.me/2L3O1f2F151c/Screen%20Shot%202017-10-20%20at%202.48.09%20PM.png)

justincampbell commented:
### Vault

        modified:   build/docs/audit/file.html
        modified:   build/docs/audit/index.html
        modified:   build/docs/audit/socket.html
        modified:   build/docs/audit/syslog.html
        modified:   build/docs/commands/environment.html
        modified:   build/intro/getting-started/apis.html

![](http://c.justincampbell.me/0L2o3y2P0L03/Screen%20Shot%202017-10-20%20at%202.55.54%20PM.png)

![](http://c.justincampbell.me/1V3t1a15091a/Screen%20Shot%202017-10-20%20at%202.57.41%20PM.png)

![](http://c.justincampbell.me/2R382s0N441m/Screen%20Shot%202017-10-20%20at%202.58.20%20PM.png)

justincampbell commented:
@sethvargo I feel pretty confident this didn't affect any of the other sites. The places where there's a diff are typically nested HTML, where the indentation does not affect rendering.

justincampbell commented:
*(Also verified I don't have that bold font locally on master)*

![](http://c.justincampbell.me/1Y1L2p040u3W/Screen%20Shot%202017-10-20%20at%203.00.24%20PM.png)

sethvargo commented:
Thanks for the rigorous testing!
